### PR TITLE
vm/credentials (search method)

### PR DESCRIPTION
**Pull request by harshshah-tenable**
Search method changes for Credentials endpoint

### CODE REVIEW

File: tenable/io/v3/vm/credentials/api.py [hunk 1]
```diff
@@ -65,7 +69,7 @@ def create(self, cred_name: str, cred_type: str,
 
         Examples:
             >>> group_id = '00000000-0000-0000-0000-000000000000'
-            >>> tio.v3.vm.credentials.create('SSH Account', 'SSH',
+            ... tio.v3.vm.credentials.create('SSH Account', 'SSH',
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/credentials/api.py]:
```suggestion
            >>> tio.v3.vm.credentials.create('SSH Account', 'SSH',
```

File: tenable/io/v3/vm/credentials/api.py [hunk 2]
```diff
@@ -132,7 +136,7 @@ def edit(self, cred_id: UUID, cred_name: Optional[str] = None,
 
         Examples:
             >>> cred_id = '00000000-0000-0000-0000-000000000000'
-            >>> tio.v3.vm.credentials.edit(cred_id,
+            ... tio.v3.vm.credentials.edit(cred_id,
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/credentials/api.py]:
```suggestion
            >>> tio.v3.vm.credentials.edit(cred_id,
```

File: tenable/io/v3/vm/credentials/api.py [hunk 3]
```diff
@@ -172,9 +176,9 @@ def details(self, id: UUID) -> Dict:
 
         Examples:
             >>> cred_id = '00000000-0000-0000-0000-000000000000'
-            >>> cred = tio.v3.vm.credentials.details(cred_id)
+            ... cred = tio.v3.vm.credentials.details(cred_id)
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/credentials/api.py]:
```suggestion
            >>> cred = tio.v3.vm.credentials.details(cred_id)
```

File: tenable/io/v3/vm/credentials/api.py [hunk 4]
```diff
@@ -191,7 +195,7 @@ def delete(self, id: UUID) -> bool:
 
         Examples:
             >>> cred_id = '00000000-0000-0000-0000-000000000000'
-            >>> cred = tio.v3.vm.credentials.delete(cred_id)
+            ... cred = tio.v3.vm.credentials.delete(cred_id)
```
Comment (ngaglani-tenable) [inline tenable/io/v3/vm/credentials/api.py]:
```suggestion
            >>> cred = tio.v3.vm.credentials.delete(cred_id)
```

**Comment by ngaglani-tenable** [comment]
`edit` and `search` methods can be alphabetized.

**Comment by ngaglani-tenable** [comment]
Format the parameters in `create` and `edit` methods.

File: tests/io/v3/vm/credentials/test_credentials_api.py [hunk 5]
```diff
@@ -211,3 +216,75 @@ def test_upload(api):
...
+
+    iterator = api.v3.vm.credentials.search(
+        fields=fields, sort=sort, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_res['pagination']['total']
```
Comment (ngaglani-tenable) [inline tests/io/v3/vm/credentials/test_credentials_api.py]:
```suggestion
    credential_list = []
    for credential in iterator:
        credential_list.append(credential)
    assert len(credential_list) == api_res['pagination']['total']
```

**Review by ngaglani-tenable — APPROVED** [review]

File: tenable/io/v3/vm/credentials/api.py [hunk 6]
```diff
@@ -237,3 +241,93 @@ def upload(self, fobj: BinaryIO) -> str:
...
+                    ...  'and': [{
+                    ...      'value': '3',
+                    ...      'operator': 'oper',
+                    ...      'property': 3
+                    ...  }]
+                    ... }
+                As the filters may change and sortable fields may change over
+                time. Please refer to the respective definition endpoint for
+                more information.
+                For Credentaials endpoint respective definitions endpoint
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/credentials/api.py]: Typo: Credentaials  -> Credentials

File: tests/io/v3/vm/credentials/test_credentials_api.py [hunk 7]
```diff
@@ -211,3 +216,75 @@ def test_upload(api):
...
+
+    iterator = api.v3.vm.credentials.search(
+        fields=fields, sort=sort, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    credential_list = []
+    for credential in iterator:
+        credential_list.append(credential)
+    assert len(credential_list) == api_res['pagination']['total']
```
Comment (alpesh-te) [inline tests/io/v3/vm/credentials/test_credentials_api.py]: Rewrite it to get rid of loop.

File: tenable/io/v3/vm/credentials/api.py [hunk 8]
```diff
@@ -157,43 +203,102 @@ def edit(self, cred_id: UUID, cred_name: Optional[str] = None,
...
-        Args:
-            id (UUID): The UUID of the credential to retrieve.
+            Iterable:
+                The iterable that handles the pagination for the job.
 
-        Returns:
-            :obj:`bool`:
-                The status of the action.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (alpesh-te) [inline tenable/io/v3/vm/credentials/api.py]: Update docstring.

**Review by alpesh-te — CHANGES_REQUESTED** [review]
Minor fixes, other than that LGTM

File: tenable/io/v3/vm/credentials/api.py (hunk 8 above)
Comment (alpesh-te) [inline tenable/io/v3/vm/credentials/api.py]: Update docstring.

**Review by alpesh-te — CHANGES_REQUESTED** [review]
Normal fixes other than that LGTM!

File: tests/io/v3/vm/credentials/test_credentials_api.py [hunk 9]
```diff
@@ -211,3 +216,71 @@ def test_upload(api):
...
+        f'{CREDENTIALS_BASE_URL}/search',
+        json=api_res,
+        match=[matchers.json_params_matcher(api_payload)],
+    )
+
+    iterator = api.v3.vm.credentials.search(
+        fields=fields, sort=sort, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+    assert len(list(iterator)) == api_res['pagination']['total']
```
Comment (alpesh-te) [inline tests/io/v3/vm/credentials/test_credentials_api.py:276]: Rewrite it to get rid of loop.

**Review by alpesh-te — CHANGES_REQUESTED** [review]
Minor fix, other than that LGMT

**Review by alpesh-te — APPROVED** [review]

**Pull request merged**